examples/rpc/src/main.rs

```diff
@@ -0,0 +1,268 @@
+                elapsed_time(),
+                caller_identity,
+                number,
+                response_timeout_ms.as_secs()
+            );
+
+            println!("[{}] [Math Genius] *doing math*…", elapsed_time());
+            sleep(Duration::from_secs(2)).await;
+
+            let result = number.sqrt();
+            println!("[{}] [Math Genius] Aha! It's {}", elapsed_time(), result);
+            Ok(json!({"result": result}).to_string())
+        })
+    });
+
+    math_genius_room.local_participant().register_rpc_method(
+        "divide".to_string(),
+        |_, caller_identity, payload, _| {
+            Box::pin(async move {
+                let json_data: Value = serde_json::from_str(&payload).unwrap();
+                let dividend = json_data["dividend"].as_i64().unwrap();
+                let divisor = json_data["divisor"].as_i64().unwrap();
+                println!(
+                    "[{}] [Math Genius] {} wants me to divide {} by {}.",
+                    elapsed_time(),
+                    caller_identity,
+                    dividend,
+                    divisor
+                );
+
+                let result = dividend / divisor;
+                println!("[{}] [Math Genius] The result is {}", elapsed_time(), result);
+                Ok(json!({"result": result}).to_string())
+            })
+        },
+    );
+}
+
+async fn perform_greeting(room: &Arc<Room>) -> Result<(), Box<dyn std::error::Error>> {
+    println!("[{}] Letting the greeter know that I've arrived", elapsed_time());
+    match room
+        .local_participant()
+        .perform_rpc("greeter".to_string(), "arrival".to_string(), "Hello".to_string(), None)
+        .await
+    {
+        Ok(response) => {
+            println!("[{}] That's nice, the greeter said: \"{}\"", elapsed_time(), response)
+        }
+        Err(e) => println!("[{}] RPC call failed: {:?}", elapsed_time(), e),
+    }
+    Ok(())
+}
+
+async fn perform_square_root(room: &Arc<Room>) -> Result<(), Box<dyn std::error::Error>> {
+    println!("[{}] What's the square root of 16?", elapsed_time());
+    match room
+        .local_participant()
+        .perform_rpc(
+            "math-genius".to_string(),
+            "square-root".to_string(),
+            json!({"number": 16}).to_string(),
+            None,
+        )
+        .await
+    {
+        Ok(response) => {
+            let parsed_response: Value = serde_json::from_str(&response)?;
+            println!("[{}] Nice, the answer was {}", elapsed_time(), parsed_response["result"]);
+        }
+        Err(e) => log::error!("[{}] RPC call failed: {:?}", elapsed_time(), e),
+    }
+    Ok(())
+}
+
+async fn perform_quantum_hypergeometric_series(
+    room: &Arc<Room>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    println!("[{}] What's the quantum hypergeometric series of 42?", elapsed_time());
```

Comment:
What the heck :) 

<img width="867" alt="image" src="https://github.com/user-attachments/assets/f53bfc94-71d1-49ac-b284-d715898a68f7">

Reply:
👋🏻